bugbug/feature_cleanup.py

```diff
@@ -0,0 +1,177 @@
+        "api-ms-win-core-string-l1-1-0.dll",
+        "api-ms-win-core-synch-l1-1-0.dll",
+        "api-ms-win-core-synch-l1-2-0.dll",
+        "api-ms-win-core-sysinfo-l1-1-0.dll",
+        "api-ms-win-core-timezone-l1-1-0.dll",
+        "api-ms-win-core-util-l1-1-0.dll",
+        "api-ms-win-crt-conio-l1-1-0.dll",
+        "api-ms-win-crt-convert-l1-1-0.dll",
+        "api-ms-win-crt-environment-l1-1-0.dll",
+        "api-ms-win-crt-filesystem-l1-1-0.dll",
+        "api-ms-win-crt-heap-l1-1-0.dll",
+        "api-ms-win-crt-locale-l1-1-0.dll",
+        "api-ms-win-crt-math-l1-1-0.dll",
+        "api-ms-win-crt-multibyte-l1-1-0.dll",
+        "api-ms-win-crt-private-l1-1-0.dll",
+        "api-ms-win-crt-process-l1-1-0.dll",
+        "api-ms-win-crt-runtime-l1-1-0.dll",
+        "api-ms-win-crt-stdio-l1-1-0.dll",
+        "api-ms-win-crt-string-l1-1-0.dll",
+        "api-ms-win-crt-time-l1-1-0.dll",
+        "api-ms-win-crt-utility-l1-1-0.dll",
+        "d3dcompiler_47.dll",
+        "freebl3.dll",
+        "mozavcodec.dll",
+        "mozavutil.dll",
+        "mozglue.dll",
+        "msvcp140.dll",
+        "nss3.dll",
+        "nssckbi.dll",
+        "nssdbm3.dll",
+        "qipcap64.dll",
+        "softokn3.dll",
+        "ucrtbase.dll",
+        "vcruntime140.dll",
+        "xul.dll",
+        "clearkey.dll",
+        "libfreebl3.dylib",
+        "liblgpllibs.dylib",
+        "libmozavcodec.dylib",
+        "libmozavutil.dylib",
+        "libmozglue.dylib",
+        "libnss3.dylib",
+        "libnssckbi.dylib",
+        "libnssdbm3.dylib",
+        "libplugin_child_interpose.dylib",
+        "libsoftokn3.dylib",
+    ]
+).replace(".", r"\.")
+
+
+def dll(text):
+    regex = fr"\b(?!{FIREFOX_DLLS_MATCH})\w+(\.dll|\.so|\.dylib)\b"
+    return re.sub(regex, "__DLL_NAME__", text)
+
+
+def synonyms(text):
+    synonyms = [
+        ("safemode", ["safemode", "safe mode"]),
+        ("str", ["str", "steps to reproduce", "repro steps"]),
+        ("uaf", ["uaf", "use after free", "use-after-free"]),
+        ("asan", ["asan", "address sanitizer", "addresssanitizer"]),
+        (
+            "permafailure",
+            [
+                "permafailure",
+                "permafailing",
+                "permafail",
+                "perma failure",
+                "perma failing",
+                "perma fail",
+                "perma-failure",
+                "perma-failing",
+                "perma-fail",
+            ],
+        ),
+        ("spec", ["spec", "specification"]),
+    ]
+
+    for synonym_group, synonym_list in synonyms:
+        text = re.sub(
+            "|".join(fr"\b{synonym}\b" for synonym in synonym_list),
+            synonym_group,
+            text,
+            flags=re.IGNORECASE,
+        )
+
+    return text
+
+
+def crash(text):
+    return re.sub(
+        r"bp-[a-f0-9]{8}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{6}[0-9]{6}\b",
+        "__CRASH_STATS_LINK__",
+        text,
+    )
```

Comment:
You might wants to pre-compile the regex if those functions are gonna be called a lot

Reply:
Good idea, I'll file a follow-up for this though (as in this PR I've just moved them from one file to another).

Reply:
Filed https://github.com/mozilla/bugbug/issues/338.